src/components/MapView.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import ReactMapboxGl, {Layer, Feature, Popup} from 'react-mapbox-gl';
+import axios from "axios";
+import { Map } from 'immutable';
+import PointTooltip from '../components/PointTooltip';
+
+const Map2 = ReactMapboxGl({
+    accessToken: process.env.REACT_APP_MAPBOX_TOKEN
+});
+
+const speciesColorMap = Map({'black bear':'#801e78','bobcat': '#498029','coyote': '#561480','cougar/mountain lion': '#802a00','raccoon': '#093c80','opossum': '#FFDC26','river otter': '#7083ff'});
+const getReports = "https://us-central1-seattlecarnivores-edca2.cloudfunctions.net/getReports"
+
+class MapView extends Component {
+    state = {
+        viewport: {
+            center: [-122.335167, 47.608013],
+            zoom: [10],
+        },
+        popupInfo: false,
+        reports: null,
+    };
+
+    componentDidMount() {
+        const url = getReports+"?mapLat=47.608013&mapLng=-122.335167";
+        axios.get(url)
+            .then(reports => {
+                this.setState({ reports: reports.data});
+            })
+            .catch(error => error);
+    }
+
+    onMoveEnd = e => {
+        let center = e.getCenter();
+        const url = getReports+"?mapLat="+center.lat+"&mapLng="+center.lng;
+        axios.get(url)
+            .then(reports => {
+                console.log(reports.data)
+                if(reports.data!=="No data!")
+                {
+                    this.setState({ reports: reports.data});
+                }
+            })
+            .catch(error => error);
+
+    };
+
+    getColor(species) {
+        return speciesColorMap.get(species)!== undefined ? speciesColorMap.get(species) : '#805b14';
+    }
+
+    renderPopup() {
+        const {popupInfo} = this.state;
+
+        return popupInfo && (
```

Comment:
Mmm is this `popupInfo &&` checking that the popupInfo is defined before we try and render the content?

Reply:
Yes! I just want to render it if we have the content. Do you think there is a better way to do it?

Reply:
Yeah! So I typically use something like this:

` if (!reports) {
      return <CircularProgress/>;
    }`

This checks to see if the reports are defined. If not, we just render a circular spinner. Once the content populates, the component automatically re-renders and we can render the actual content.

Reply:
will update it! Thank you!